docs/ZXing_barcode.md

```diff
@@ -0,0 +1,73 @@
+
+**Entry Point to the Code**
+
+For developers new to a code base, often being pointed to where user interaction first interfaces with the code base is a good place to start. In the case of the barcode scanner functionality, this can be found within the UploadDocumentForm.jsx component within the src/components directory:
+
+```
+src/components/Form/UploadDocumentForm.jsx
+```
+
+At the time of this writing, the line specific to uploading files can be found on line 67 of this file, where the entry point of what happens when the user clicks on the "Upload file", as this triggers the handleFormSubmission function, defined on line 48. This function in turn calls the makeHandleFormSubmission function found within the utils directory, where the majority of the barcode scanner logic lives:
+
+```
+src/utils/frontend/FormSubmissionHelper.js
+```
+
+This function handles a good deal of logic for submitting a wide variety of forms to the Solid Server, including parsing the fileObject passed based off the event, and most importantly uploading the document, specifically by calling the uploadDocument() function defined on line 47. Going to the definition of this function leads us to:
+
+```
+src/utils/network/session-core.js
+```
+
+Defined on line 162, uploadDocument() contains a great deal of logic specific to solid and we start to see a good deal of functions coming from the [@inrupt/solid-client libraries](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/solid-client/). Of particular interest to the subject of this document is the createResourceTtlFile() function on line 191, which is defined within:
+
+```
+src/utils/network/session-helper.js
+```
+
+Here on line 346 we see the logic specific to @inrupt/solid-client, which takes our fileObject and documentUrl and builds a generic .ttl file to be sent to the Solid server. Specific to the barcode scanner is the if statement check on line 349 which checks to see if the user has selected the "Driver's License" document from the "File to upload:" in the UploadDocumentForm.jsx component. Upon inspection of this if statement, we see that if the user has selected "Driver's License", the function returns the value of a separate helper function called createDriversLicenseTtlFile(), defined on line 300. This lengthy function is very similar to the createResourceTtlFile() function, but starts itself with a helper function of its own called getDriversLicenseData, which takes the fileObject.file as an argument. This helper function is defined in:
```

Comment:
this line number is not correct anymore

Reply:
Going to using references rather than line numbers for an update.